File: erettsegi_emelt/2018_okt/c#/Utca.cs
using System;
using System.Collections.Generic;
using System.IO;

var sorok = File.ReadAllLines("kerites.txt");
var telkek = new List<Telek>();

var parosHazszam = 2;
var paratlanHazszam = 1;

foreach(var sor in sorok) {
    var split = sor.Split(' ');
    var parose = int.Parse(split[0]) == 0;

    telkek.Add(new Telek(parose, parose ? parosHazszam : paratlanHazszam, int.Parse(split[1]), split[2][0]));

    if(parose) {
        parosHazszam += 2;
    }else{
        paratlanHazszam += 2;
    }
}

Console.WriteLine("2. Feladat:");
Console.WriteLine($"Eladott telkek száma: {telkek.Count}");

var utolsoTelek = telkek[telkek.Count - 1];

Console.WriteLine("3. Feladat:");
Console.WriteLine($"Az utolsó telek: {(utolsoTelek.parosE ? "Páros" : "Páratlan")}");
Console.WriteLine($"Az utolsó telek házszáma: {utolsoTelek.hazszam}");
Console.WriteLine("4. Feladat:");

for(var i = 0; i < telkek.Count - 1; ++i) {
    var telek = telkek[i];
    var kovetkezo = telkek[i + 1];

    if (!telek.parosE && telek.keritesSzine != ':' && telek.keritesSzine != '#' && !kovetkezo.parosE && kovetkezo.keritesSzine == telek.keritesSzine) {
        Console.WriteLine($"Talált házszám: {telek.hazszam}");
        break;
    }
}

Console.WriteLine("5. Feladat:");
Console.WriteLine("Írd be 1 telek számát!");
var beolvasottTelekSzam = int.Parse(Console.ReadLine());

for(int i = 0; i < telkek.Count; ++i) {
    var jelenlegiTelek = telkek[i];

    if(jelenlegiTelek.hazszam == beolvasottTelekSzam) {
        Console.WriteLine("Kerítés színe: " + (jelenlegiTelek.keritesSzine == ':' ? "Nem készült el" : jelenlegiTelek.keritesSzine == '#' ? "Festetlen" : jelenlegiTelek.keritesSzine.ToString()));

        var balSzomszed = telkek[i - 1];
        var jobbSzomszed = telkek[i + 1];

        for(char generalt = 'A'; ; ++generalt) {
            if(balSzomszed.keritesSzine != generalt && jobbSzomszed.keritesSzine != generalt && jelenlegiTelek.keritesSzine != generalt) {
                Console.WriteLine($"Az új szín lehet: {generalt}");
                break;
            }
        }

        break;
    }
}

using var output = new StreamWriter("utcakep.txt");

foreach(var telek in telkek) {
    if(!telek.parosE) {
        for(int k = 0; k < telek.szelesseg; ++k) {
            output.Write(telek.keritesSzine);
        }
    }
}

output.WriteLine();

foreach(var telek in telkek) {
    if (!telek.parosE) {
        for(int k = 0; k < telek.szelesseg; ++k) {
            if(k == 0) {
                output.Write(telek.hazszam);

                if(telek.hazszam > 9) {
                    ++k;
                }

                if(telek.hazszam > 99) {
                    ++k;
                }
            }else{
                output.Write(' ');
            }
        }
    }
}

File: erettsegi_emelt/2018_okt/java/Utca.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Utca {

    public static void main(String[] args) throws IOException {
        var lines = Files.readAllLines(Path.of("kerites.txt"));
        var telkek = new ArrayList<Telek>();

        var parosHazszam = 2;
        var paratlanHazszam = 1;

        for(var line : lines) {
            var split = line.split(" ");
            var parose = Integer.parseInt(split[0]) == 0;

            if(parose) {
                telkek.add(new Telek(parosHazszam, Integer.parseInt(split[1]), split[2].charAt(0)));
                parosHazszam += 2;
            }else{
                telkek.add(new Telek(paratlanHazszam, Integer.parseInt(split[1]), split[2].charAt(0)));
                paratlanHazszam += 2;
            }
        }

        System.out.println("2. Feladat:");
        System.out.println("Eladott telkek száma: " + telkek.size());

        var utolsoTelek = telkek.get(telkek.size() - 1);

        System.out.println("3. Feladat:");
        System.out.println("Az utolsó telek: " + (utolsoTelek.parosE ? "Páros" : "Páratlan"));
        System.out.println("Az utolsó telek házszáma: " + utolsoTelek.hazszam);
        System.out.println("4. Feladat:");

        for(var i = 0; i < telkek.size() - 1; ++i){
            var telek = telkek.get(i);
            var kovetkezo = telkek.get(i + 1);

            if(!telek.parosE && telek.keritesSzine != ':' && telek.keritesSzine != '#' && !kovetkezo.parosE && kovetkezo.keritesSzine == telek.keritesSzine) {
                System.out.println("Talált házszám: " + telek.hazszam);
                break;
            }
        }

        System.out.println("5. Feladat:");
        System.out.println("Írd be 1 telek számát!");

        try(var console = new Scanner(System.in)) {
            var beolvasottTelekSzam = console.nextInt();

            for(var i = 0; i < telkek.size(); ++i){
                var jelenlegiTelek = telkek.get(i);

                if(jelenlegiTelek.hazszam == beolvasottTelekSzam) {
                    System.out.println("Kerítés színe: " + (jelenlegiTelek.keritesSzine == ':' ? "Nem készült el" : jelenlegiTelek.keritesSzine == '#' ? "Festetlen" : jelenlegiTelek.keritesSzine));

                    var balSzomszed = telkek.get(i - 1);
                    var jobbSzomszed = telkek.get(i + 1);

                    for(var generalt = 'A'; ; ++generalt) {
                        if(balSzomszed.keritesSzine != generalt && jobbSzomszed.keritesSzine != generalt && jelenlegiTelek.keritesSzine != generalt) {
                            System.out.println("Az új szín lehet: " + generalt);
                            break;
                        }
                    }
                    break;
                }
            }
        }

        try(var output = new PrintWriter("utcakep.txt")){
            for(var telek : telkek) {
                if(!telek.parosE) {
                    for(var k = 0; k < telek.szelesseg; ++k) {
                        output.print(telek.keritesSzine);
                    }
                }
            }

            output.println();

            for(var telek : telkek) {
                if(!telek.parosE) {
                    var szokozSzam = telek.szelesseg - Integer.toString(telek.hazszam).length();

                    output.print(telek.hazszam + " ".repeat(szokozSzam));
                }
            }
        }
    }
}